packages/service-worker/src/services/accounts/AccountsService.ts

```diff
@@ -516,6 +569,18 @@ export class AccountsService implements OnLock, OnUnlock {
       windowId: crypto.randomUUID(),
       properties: { addresses: await this.#getAllAddresses() },
     });
+
+    if (addAllWithHistory) {
+      const hasNextAccountsHistory = await this.#hasNextAccountsHistory({
+        walletId,
+        index: nextIndex + 1,
+        accountsCount: 2,
+      });
+      if (hasNextAccountsHistory) {
+        return this.addPrimaryAccount({ walletId, addAllWithHistory: true });
```

Comment:
I find the recursive logic to find the addresses a bit too difficult to wrap my head around. Feels like this could simply just be a single `while` loop

```
let lastIndexWithActivity = 0;
let gapWithoutActivity = 0;
while(gapWithoutActivity < 2 && lastIndexWithActivity < MAX_NUMBER_OF_ACCOUNTS) {
    
	// fetch activity
	if(!activity) {
		gapWithoutActivity++;
		continue;
	}

	lastIndexWithActivity++;
	gapWithoutActivity = 0;
	
	#add account
}
```